control-admin/src/main/java/fi/nls/oskari/control/admin/GetAllRolesAndPermissionTypesHandler.java

```diff
@@ -0,0 +1,121 @@
+package fi.nls.oskari.control.admin;
+
+import static fi.nls.oskari.control.ActionConstants.KEY_ID;
+import static fi.nls.oskari.control.ActionConstants.KEY_NAME;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.oskari.permissions.PermissionService;
+import org.oskari.permissions.model.PermissionType;
+
+import fi.nls.oskari.annotation.OskariActionRoute;
+import fi.nls.oskari.control.ActionException;
+import fi.nls.oskari.control.ActionParameters;
+import fi.nls.oskari.control.ActionParamsException;
+import fi.nls.oskari.control.RestActionHandler;
+import fi.nls.oskari.domain.Role;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.service.OskariComponentManager;
+import fi.nls.oskari.service.ServiceException;
+import fi.nls.oskari.service.UserService;
+import fi.nls.oskari.util.JSONHelper;
+import fi.nls.oskari.util.ResponseHelper;
+
+@OskariActionRoute("GetAllRolesAndPermissionTypes")
+public class GetAllRolesAndPermissionTypesHandler extends RestActionHandler {
+    
+    private Logger log = LogFactory.getLogger(GetAllRolesAndPermissionTypesHandler.class);
+    private UserService userService = null;
+    private PermissionService permissionsService = null;
+
+    private final static String JSKEY_ID = "id";
+    private final static String JSKEY_NAME = "name";
+    private final static String JSKEY_ROLES = "roles";
+    private final static String JSKEY_PERMISSION_TYPES = "permissionTypes";
+
+    @Override
+    public void init() {
+        try {
+            userService = UserService.getInstance();
+        } catch (Exception ex) {
+            log.error(ex, "Unable to initialize User service!");
+        }
+        try {
+            permissionsService = OskariComponentManager.getComponentOfType(PermissionService.class);
+        } catch (Exception ex) {
+            log.error(ex, "Unable to initialize permission service!");
```

Comment:
Since we never try to recover from the error later on it would be best to let exceptions be thrown from the init(). That way the ActionHandler is not available for use by the frontend and we don't need to check if the services are initialized correctly for every request on preProcess().

Reply:
I now fixed this such way that checked exceptions are converted to unchecked RuntimeExceptions and those are thrown. In this way we don't need to add checked exception(s) to init method signatures of parent classes and cause need to make changes to other action handlers.

Reply:
Remove the log.error() calls as well so we don't get the exception logged twice.

Reply:
You can also use ServiceRuntimeException instead of RuntimeException to notify it's a exception from Oskari instead of a plain runtime one.

Reply:
Fixed as proposed